WordPress/src/main/java/org/wordpress/android/models/Note.kt

```diff
@@ -0,0 +1,525 @@
+     */
+    fun canModerate() = enabledCommentActions.contains(EnabledActions.ACTION_APPROVE) ||
+            enabledCommentActions.contains(EnabledActions.ACTION_UNAPPROVE)
+
+    fun canReply() = enabledCommentActions.contains(EnabledActions.ACTION_REPLY)
+
+    fun canLikeComment() = enabledCommentActions.contains(EnabledActions.ACTION_LIKE_COMMENT)
+
+    fun canLikePost() = enabledPostActions.contains(EnabledActions.ACTION_LIKE_POST)
+
+    var localStatus: String?
+        get() = StringUtils.notNullStr(mLocalStatus)
+        set(localStatus) {
+            mLocalStatus = localStatus
+        }
+
+    @Suppress("SwallowedException")
+    val subject: JSONObject?
+        get() {
+            try {
+                synchronized(mSyncLock) {
+                    val subjectArray = mNoteJSON?.getJSONArray("subject")
+                    if (subjectArray != null && subjectArray.length() > 0) {
+                        return subjectArray.getJSONObject(0)
+                    }
+                }
+            } catch (e: JSONException) {
+                return null
+            }
+            return null
+        }
+
+    fun getFormattedSubject(notificationsUtilsWrapper: NotificationsUtilsWrapper): Spannable {
+        return subject?.let { notificationsUtilsWrapper.getSpannableContentForRanges(it) } ?: SpannableString("")
+    }
+
+    val title: String
+        get() = queryJSON("title", "")
+    val iconURL: String
+        get() = queryJSON("icon", "")
+    val iconURLs: List<String>?
+        get() {
+            synchronized(mSyncLock) {
+                val bodyArray = mNoteJSON?.optJSONArray("body")
+                if (bodyArray != null && bodyArray.length() > 0) {
+                    val iconUrls = ArrayList<String>()
+                    for (i in 0 until bodyArray.length()) {
+                        val iconUrl = JSONUtils.queryJSON(bodyArray, "body[$i].media[0].url", "")
+                        if (iconUrl != null && !iconUrl.isEmpty()) {
+                            iconUrls.add(iconUrl)
+                        }
+                    }
+                    return iconUrls
+                }
+            }
+            return null
+        }
+    val commentSubject: String?
+        get() {
+            synchronized(mSyncLock) {
+                val subjectArray = mNoteJSON?.optJSONArray("subject")
+                if (subjectArray != null) {
+                    var commentSubject = JSONUtils.queryJSON(subjectArray, "subject[1].text", "")
+
+                    // Trim down the comment preview if the comment text is too large.
+                    if (commentSubject != null && commentSubject.length > MAX_COMMENT_PREVIEW_LENGTH) {
+                        commentSubject = commentSubject.substring(0, MAX_COMMENT_PREVIEW_LENGTH - 1)
+                    }
+                    return commentSubject
+                }
+            }
+            return ""
+        }
+    @Suppress("SwallowedException")
```

Comment:
We seem to have a lot of swallowed exception suppressions. I'm not sure how that rule is tuned, but it seems this behavior is intentional - so I'm not sure this rule is helpful here. That said, I wonder if it'd read cleaner to use `runCatching` (perhaps chained with `getOrElse`) 🤔 . 

Maybe this is moot if we are moving to Gson, or something similar. 🤷‍♂️

Reply:
That's a good idea @mkevins 👍 
I've changed this with 73ec67c4214fcd42ec61645d55ebc6c9dea163ac